### ISSUE TITLE
[SOLVED] Update jobs "queued" with unknown status (Pulling proposals)

### ISSUE THREAD

**Issue by Tux-Panik** [MISP/MISP#2035]
Dear,
I've an issue with server/feeds updates.
All jobs (scheduled or not) stay in "queued" status and MISP didn't be updated.
Updates worked until 06/Mar/2017. This day, I had a job blocked at 10%.
Since this day, I didn't set anything, except scheduled pull tasks every 6 hours. 

I made the following operations to troubleshoot problem:

1/ Stop all queue's jobs
redis-cli
flushdb
flushall
quit
su -s /bin/bash apache -c '/usr/bin/scl enable rh-php56 /var/www/MISP/app/Console/worker/start.sh'
 

2/ Clear all queue's jobs
mysql -u <user> -p
use misp
delete from jobs;
exit

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | support
| OS version (server)       | CentOS
| OS version (client)       | 7
| PHP version               | 5.6
| MISP version / git hash   | 2.4.65
| Browser                   | FF

### Expected behavior
To reach MISP updates from servers & OSINT feeds.

### Actual behavior
All jobs are queued, with "Pulling proposals." message.
Workers are OK and jobs in queue are all at zero.

### Steps to reproduce the behavior
NA

### Logs, screenshots, configuration dump, ...
cat /var/www/MISP/app/tmp/logs/resque-scheduler-2017-03-09.log
[2017-03-09 00:00:02] main.INFO: Moving scheduled job SERVERSHELL to DEFAULT {"type":"movescheduled","args":{"timestamp":1488931200,"class":"ServerShell","queue":"default","job_id":"37545bb97462a9c04d0809c1df5363af","wait":1488931202.143,"s_wait":1488931200},"worker":"misp:13893"} []
[2017-03-09 06:00:01] main.INFO: Moving scheduled job SERVERSHELL to DEFAULT {"type":"movescheduled","args":{"timestamp":1488952800,"class":"ServerShell","queue":"default","job_id":"4a741161000b22a62907f69b4cbba56c","wait":1488952801.861,"s_wait":1488952800},"worker":"misp:13893"} []
[2017-03-09 12:00:01] main.INFO: Moving scheduled job SERVERSHELL to DEFAULT {"type":"movescheduled","args":{"timestamp":1488974400,"class":"ServerShell","queue":"default","job_id":"bedf10f1ed50190e8fd992e82a7469fe","wait":1488974401.459,"s_wait":1488974400},"worker":"misp:13893"} []
[2017-03-09 18:00:01] main.INFO: Moving scheduled job SERVERSHELL to DEFAULT {"type":"movescheduled","args":{"timestamp":1488996000,"class":"ServerShell","queue":"default","job_id":"3dc009b5f4732127a14f54086f7248ff","wait":1488996001.033,"s_wait":1488996000},"worker":"misp:13893"} []

cat /var/www/MISP/app/tmp/logs/resque-2017-03-09.log
[2017-03-09 00:00:03] main.INFO: got {"queue":"default","id":"37545bb97462a9c04d0809c1df5363af","class":"ServerShell","args":[{"0":"enqueuePull","1":1488931200,"2":"1","3":"2","s_time":1488909604}]} {"type":"got","args":"[object] (Resque_Job: {\"queue\":\"default\",\"id\":\"37545bb97462a9c04d0809c1df5363af\",\"class\":\"ServerShell\",\"args\":[{\"0\":\"enqueuePull\",\"1\":1488931200,\"2\":\"1\",\"3\":\"2\",\"s_time\":1488909604}]})","worker":"misp:13821"} []
[2017-03-09 00:00:03] main.INFO: Processing ID:37545bb97462a9c04d0809c1df5363af in default {"type":"process","worker":"misp:13821","job_id":"37545bb97462a9c04d0809c1df5363af"} []
[2017-03-09 06:00:03] main.INFO: got {"queue":"default","id":"4a741161000b22a62907f69b4cbba56c","class":"ServerShell","args":[{"0":"enqueuePull","1":1488952800,"2":"1","3":"2","s_time":1488931203}]} {"type":"got","args":"[object] (Resque_Job: {\"queue\":\"default\",\"id\":\"4a741161000b22a62907f69b4cbba56c\",\"class\":\"ServerShell\",\"args\":[{\"0\":\"enqueuePull\",\"1\":1488952800,\"2\":\"1\",\"3\":\"2\",\"s_time\":1488931203}]})","worker":"misp:13821"} []
[2017-03-09 06:00:03] main.INFO: Processing ID:4a741161000b22a62907f69b4cbba56c in default {"type":"process","worker":"misp:13821","job_id":"4a741161000b22a62907f69b4cbba56c"} []
[2017-03-09 12:00:02] main.INFO: got {"queue":"default","id":"bedf10f1ed50190e8fd992e82a7469fe","class":"ServerShell","args":[{"0":"enqueuePull","1":1488974400,"2":"1","3":"2","s_time":1488952803}]} {"type":"got","args":"[object] (Resque_Job: {\"queue\":\"default\",\"id\":\"bedf10f1ed50190e8fd992e82a7469fe\",\"class\":\"ServerShell\",\"args\":[{\"0\":\"enqueuePull\",\"1\":1488974400,\"2\":\"1\",\"3\":\"2\",\"s_time\":1488952803}]})","worker":"misp:13821"} []
[2017-03-09 12:00:02] main.INFO: Processing ID:bedf10f1ed50190e8fd992e82a7469fe in default {"type":"process","worker":"misp:13821","job_id":"bedf10f1ed50190e8fd992e82a7469fe"} []
[2017-03-09 18:00:04] main.INFO: got {"queue":"default","id":"3dc009b5f4732127a14f54086f7248ff","class":"ServerShell","args":[{"0":"enqueuePull","1":1488996000,"2":"1","3":"2","s_time":1488974402}]} {"type":"got","args":"[object] (Resque_Job: {\"queue\":\"default\",\"id\":\"3dc009b5f4732127a14f54086f7248ff\",\"class\":\"ServerShell\",\"args\":[{\"0\":\"enqueuePull\",\"1\":1488996000,\"2\":\"1\",\"3\":\"2\",\"s_time\":1488974402}]})","worker":"misp:13821"} []
[2017-03-09 18:00:04] main.INFO: Processing ID:3dc009b5f4732127a14f54086f7248ff in default {"type":"process","worker":"misp:13821","job_id":"3dc009b5f4732127a14f54086f7248ff"} []

tail resque-worker-error.log
2017-03-09 14:41:27 Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in [/var/www/MISP/app/Model/Event.php, line 1112]
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in /var/www/MISP/app/Model/Event.php on line 1112
Fatal Error Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in [/var/www/MISP/app/Model/Event.php, line 1112]
2017-03-09 15:31:08 Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in [/var/www/MISP/app/Model/Event.php, line 1112]
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in /var/www/MISP/app/Model/Event.php on line 1112
Fatal Error Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in [/var/www/MISP/app/Model/Event.php, line 1112]

Let me know if you need more information.
Thanks in advance for your inputs.
Regards,

**Comment by adulau** [comment]
~~~~~
PHP Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 32 bytes) in 
~~~~~

It looks like you hit your memory limit in PHP. Go to your php.ini and update it to a sane value depending of your memory available. (memory_limit =)

**Comment by Tux-Panik** [comment]
Thanks Alexandre for your reply,
"limit_memory" value passed from 128MB to 512MB.
Queue jobs was cleaned, workers restarted and PHP/Apache services restarted.
**But new jobs still are queued.**

In "resque-worker-error.log", I haven't any entry since I changed value (always under monitoring).

**Comment by Tux-Panik** [comment]
I got the same message Friday afternoon, but not the same mentioned file:

`2017-03-10 14:42:15 Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 47178167 bytes) in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocketResponse.php, line 160]`

Application requests 47MB of memory size more. So, I changed 512M to 1024M in /etc/php.ini
I also changed "max_execution_time" and "max_input_time" values.
I still monitor logs.

Regards,

**Comment by iglocska** [comment]
Sound like you still only have 128M of memory available:

~~~
Allowed memory size of 134217728 bytes exhausted
~~~

Did you restart apache after making the chances?

**Comment by Tux-Panik** [comment]
Yes, using `systemctl restart httpd.service`
I'm currently changing the value of "memory_limit" in /var/www/MISP/app/Lib/cakephp/lib/Cake/Test/Case/Console/Command/TestShellTest.php.

**Comment by iglocska** [comment]
That won't do much, also, you should never change anything in cakephp.

**Comment by Tux-Panik** [comment]
Just rollbacked, it didn't help in fact.
To precise: I increased "memory_limit" in /etc/php.ini and I didn't find any other "php.ini", so I guess it is the good file.

**Comment by Tux-Panik** [comment]
Oh Wait... just found two others "php.ini":

`/etc/opt/rh/rh-php56/php.ini`
`/opt/rh/rh-php56/register.content/etc/opt/rh/rh-php56/php.ini`

Same properties and same MD5

**Comment by iglocska** [comment]
Yep, for RHEL/CentOS it's /etc/opt/rh/rh-php56/php.ini that you want to edit.

**Comment by Tux-Panik** [comment]
It seems updates works when I change the second one (/opt/rh/rh-php56/register.content/etc/opt/rh/rh-php56/php.ini).
I'll confirm in a few hours.
Thanks

**Comment by iglocska** [comment]
Excellent, let us know if the issue's still not gone!

**Comment by Tux-Panik** [comment]
Update is still in progress... 33%.
I close the case.
Many thanks for your support.

**Comment by iglocska** [comment]
No worries, thanks a lot for the follow-up!

**Comment by mari0d** [comment]
I had this same exact issue - any chance modifying PHP max memory can be mentioned/recommended in the CentOS install steps?

**Comment by deni3dAcc3ss** [comment]
For posterity.  I had an issue where my jobs got backed up and we're in an 'unknown' status.  I restarted my workers and some of the jobs completed.  The other jobs had re-queued, some showing 'queued' and others a percentage between  0-100%  (ususally near 50%).  In my case the cause of the choke was an email was being sent for a user who was deleted.  Not sure if this is a race condition.    I'm going to cherrypick and delete the related failed jobs from the misp SQL jobs table.

Currently installed version… **v2.4.96 (702301565bcf4ab831bffdce4dea1bf21c1aa500)** 

[2018-11-29 20:23:00] main.INFO: Processing ID:91a380fca0583f4fd93fb9e4467b1ee4 in email {"type":"process","worker":"HOSTNAME:14921","job_id":"91a380fca0583f4fd93fb9e4467b1ee4"} []
[2018-11-29 20:23:00] main.ERROR: {"queue":"email","id":"91a380fca0583f4fd93fb9e4467b1ee4","class":"EventShell","args":[[**"alertemail","2**","185673","183285",null]]} failed: **There was an error with the user account.** {"type":"fail","log":"There was an error with the user account.","job_id":"91a380fca0583f4fd93fb9e4467b1ee4","time":47,"worker":"HOSTNAME:14921"} []

**Comment by iglocska** [comment]
Indeed, what probably happened was that you had the job queued up for the user before deletion. The fact that this can stall the execution doesn't sound great, will look into a way of gracefully skipping those users.

**Comment by deni3dAcc3ss** [comment]
I agree: 
> what probably happened was that you had the job queued up for the user before deletion. 

> 

I deleted the account with ID '2' one second before the time in the log above so 20:22:59.  The reason I deleted the account which was my admin account was because I could not edit the account's profile from the 'list users' view, so I deleted and recreated the account, where then my account had a different ID in SQL.  I could login and edit. So I wondered if I could simply change the ID back to '2' with a SQL update to the table, the update worked, but I was again plagued with the inability to edit the account,  I changed the ID back to the ID from the new creation and functionality was restored. Unfortunately I cannot recall the exact conditions/experience related to my inability to access my account's profile.
